### ISSUE TITLE
The average should not be computed in L2Pool2d

### ISSUE THREAD

**Issue by fujunwei** [tensorflow/tensorflow#73742]
The [ONNX L2Pool2d](https://github.com/onnx/onnx/blob/main/docs/Operators.md#lppool), [DirectML L2 Pooling Desc](https://docs.microsoft.com/en-us/windows/win32/api/directml/ns-directml-dml_lp_pooling_operator_desc) and [CoreML's l2_pool2d](https://apple.github.io/coremltools/source/coremltools.converters.mil.mil.ops.defs.html#coremltools.converters.mil.mil.ops.defs.iOS15.pool.l2_pool) calculate the l2 pooling by the expression `Y = (X1^2 + X2^2 + ... + Xn^2) ^ (1/2)`,  but [TFLite L2_PooL2d kernel implementation](https://source.chromium.org/chromium/chromium/src/+/main:third_party/tflite/src/tensorflow/lite/kernels/internal/optimized/optimized_ops.h;l=3341?q=src%2Ftensorflow%2Flite%2Fkernels%2Finternal%2Foptimized%2Foptimized_ops.h) has the average with the count of sum elements  `Y=((X1^2 + X2^2 + ... + Xn^2)/n) ^ (1/2)`,  is it an issue of the kernel implementation?

BTW, [the kernel of l2_norm](https://source.chromium.org/chromium/chromium/src/+/main:third_party/tflite/src/tensorflow/lite/kernels/internal/optimized/optimized_ops.h;l=1424?q=L2Normalization&ss=chromium%2Fchromium%2Fsrc) also has no the average.

**Comment by Agzamxoja** [comment]
Yes, this is an issue with the TFLite L2_Pool2d kernel implementation. The correct implementation of L2 pooling should be Y = (X1^2 + X2^2 + ... + Xn^2) ^ (1/2), without averaging by the count of sum elements.

The averaging step is not part of the standard L2 pooling operation, and it changes the result. The correct implementation should only sum the squared values and then take the square root, without dividing by the count of elements.

It's good that you also noticed the same issue with the L2 norm kernel implementation. Both L2 pooling and L2 norm should follow the same formula without averaging.

This discrepancy might cause issues when converting models between different frameworks, or when trying to reproduce results from other frameworks. It's recommended to report this issue to the TFLite developers and ask them to fix the kernel implementation to match the standard L2 pooling and L2 norm formulas.

**Comment by pkgoogle** [comment]
Current PR: https://github.com/tensorflow/tensorflow/pull/74079

**Comment by fujunwei** [comment]
Thanks, what's the status of this PR, does it have any blocking issue?

**Comment by fujunwei** [comment]
> Current PR: [#74079](https://github.com/tensorflow/tensorflow/pull/74079)

@pkgoogle  There is no reviewer for this PR, could you find someone you're familiar with to review this code?

**Comment by aravindmurali22** [comment]
@suleshahid , could you please take a look into this PR?

**Comment by aravindmurali22** [comment]
@fujunwei , Thank you for filing the issue. Since this issue was opened, many updates have been released that may already address your concern; please test the latest version to see if the problem is resolved. We also recommend trying LiteRT, as active development has transitioned there to include the latest features and fixes. If the issue persists, please open a new report in the LiteRT repository so we can investigate it within our current development workflow.
